### ISSUE TITLE
modify path arguments to new_model()

### ISSUE THREAD

**Issue by copernican** [metrumresearchgroup/rbabylon#179]
# Summary

#141 requires that the model file, YAML file, and output directory all have the same path (minus file extensions). #182 adds this path to the model object as `absolute_model_path`. In addition, we will deprecate the `.directory` argument as part of a broader refactor. `new_model()` should reflect these changes, which involves

- removing the `.directory` argument
- renaming the `.yaml_path` argument to `.path`

We also take this opportunity to ensure that `.path` can contain periods.

# Technical specification

## new_model()

1. Remove `.directory` argument, and remove call to `combine_directory_path()`.

2. Rename `.yaml_path` argument to `.path`.

3. Remove warning if `.path` (formerly, `.yaml_path`) does not have a `yaml` file extension, because `.path` will not have an extension.

4. Check that the YAML file implied by `.path` does not exist, and throw an error if it does and `overwrite = FALSE`. The error should (continue to) inform the user that this check can be skipped if `.overwrite = TRUE`.

5. Construct the absolute model path as 

    ```r
    file.path(normalizePath(dirname(.path)), basename(.path)
    ```

    to avoid a warning from `normalizePath()`, and assign the result to `.mod[[ABS_MOD_PATH]]`.

6. Change `.start` argument to `safe_based_on()` to be `dirname(.path)`.

## copy_nonmem_model_from()

1. Modify the call to `new_model()` by

    a. changing `.yaml_path` to `.path = .new_model`

    b. removing the `.directory` argument

## get_yaml_path.bbi_nonmem_model()

1. We want to allow for periods in the `.path` argument to `new_model()`. `new_model()` calls `create_model_object()`, which calls `save_model_yaml()`, which calls `get_yaml_path()`. `get_yaml_path()` calls `yaml_ext()`, but `yaml_ext("foo.bar")` will return `foo.yaml`. Thus, to allow periods in `.path`, we need to modify `get_yaml_path.bbi_nonmem_model()` to append the `yaml` extension to  `.bbi_object[[ABS_MOD_PATH]]` rather than allowing `yaml_ext()` to replace the "extension" (anything after the last period).

## find_nonmem_model_file_path()

1. Similar to `yaml_ext()`, `ctl_ext()` and `mod_ext()` will replace anything after the last period. To support periods in `.path`, we need to append `.ctl` or `.mod` to the `.path` argument rather than calling `ctl_ext()` or `mod_ext()`.

# Test changes

1. Modify tests that call `new_model()` so that they pass an appropriate `.path`.

2. Add a test that `new_model()` supports a `.path` that contains periods.

3. Modify a test of `submit_models()` that calls `copy_model_from()` where `.new_model` has an extension (this was already not supported, but passed because the extension was removed elsewhere).

4. Update vignettes that call `new_model()`.

**Comment by seth127** [comment]
Some changes and/or comments:

> 1. A warning will not be raised if .path (formerly, .yaml_path) does not have a file extension...

What should we do here? It seems like there are some options:
1. enforce that `.path` does _not_ have an extension
2. strip off any extension that `.path` has and apply the relevant extensions later
  a. do this silently, implicitly saying "pass whatever you extension you want and we'll figure it out"
  b. do this with a warning saying "`.path` should have no extension, we are inferring you mean ..."

I'm inclined towards 2b but I'd be fine with 1 as well. I think 2a sounds dangerous.

------

> 2. Check that /path/to/foo/bar exists and is a directory, and throw an error if not. This is effectively a check that the model output directory exists, not that that the model file or YAML file exists (two other meanings of .path).

This is wrong. The output directory will not exist when `new_model()` is called.

-----

> 5. Pass .out_path = "/path/to/foo/bar" to save_model_yaml().

I think we should probably just deprecate the `.out_path()` argument to `save_model_yaml()`. That was a holdover from wanting to be able to save the model yaml out to somewhere _other than_ where the model object expects it to be. This is exactly the kind of thing we are prohibiting in #141. Perhaps this change should even be made as part of #141.

**Comment by seth127** [comment]
Also, I think we should have a discussion about points 4 and 6 above. This also relates to #141. The more I think about it, the more I feel like we should only carry about a single path and then have the helper construct their other paths as needed. This simplifies the checking and makes it harder for a user to shoot themselves in the foot by changing one path but not others.

**Comment by copernican** [comment]
Discussed over Slack, and agreed on the following:

1. #141 describes throwing an error if the model file does not exist. For `new_model()`, we will require that the model file exist at `.path`, e.g., if `.path = "foo/bar"`, then one of `foo/bar.ctl` or `foo/bar.mod` must exist (though the actual check will be done by a helper function that contains the specific logic).

2. Neither the output directory nor the YAML file is expected to exist when `new_model()` is called. In particular, `new_model()` is responsible for _creating_ the YAML file, and the output directory will be created by `submit_model()`, which is necessarily called after model creation.

3. #141 removes the need to have an argument to `save_model_yaml()` like `.out_path`, because the YAML file will be required to be in the same directory as the model file. We will modify `save_model_yaml()` accordingly.

4. We will resolve which paths are included in the model object in #182, and the technical specification for this PR will reflect those decisions.
